### ISSUE TITLE
SDP function for different chip size

### ISSUE THREAD

**Issue by elrie** [oddblk/eeprom-writer#3]
Hi!

For different chip size different addresses have to be used for SDP enable/disable regarding to datasheets.
Now there are hardcoded addresses only for 28C64

For example for AT28C256 code should look like:

```
  if (bWriteProtect)
  {
    WriteByteTo(0x5555, 0xAA);
    WriteByteTo(0x2AAA, 0x55);
    WriteByteTo(0x5555, 0xA0);
  }
  else
  {
    WriteByteTo(0x5555, 0xAA);
    WriteByteTo(0x2AAA, 0x55);
    WriteByteTo(0x5555, 0x80);
    WriteByteTo(0x5555, 0xAA);
    WriteByteTo(0x2AAA, 0x55);
    WriteByteTo(0x5555, 0x20);
  }

```

**Comment by JOliverasC** [comment]
Yes, these addresses work the same for smaller (less capacity) EEPROMS. They lost the most significant bits because the EEPROM have not upper address lines)  
I put these changes in the branch https://github.com/JOliverasC/eeprom-writer with other changes (Chip erase by software and Page Write mode for memories supporting those features) 
Also there is a clone of Windows utility for EEPROM Writer http://danceswithferrets.org/geekblog/?p=924  with source code at https://github.com/JOliverasC/-eewriter

**Comment by lazuli2393** [comment]
I wish I had looked here before spending hours debugging.
I can confirm, for the AT28C256-15PC, the Software Protection code is different.
    WriteByteTo(0x5555, 0xAA);
    WriteByteTo(0x2AAA, 0x55); 
    WriteByteTo(0x5555, 0x80);
    WriteByteTo(0x5555, 0xAA);
    WriteByteTo(0x2AAA, 0x55);
    WriteByteTo(0x5555, 0x20);

**Comment by ghost** [comment]
Thank you so much for this.
